pkg/landscaper/gardenlet/applier/common/rbac.go

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"context"
+
+	. "github.com/onsi/gomega"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+)
+
+// ValidateGardenletRBACChartResources validates that the resources of the Gardenlet RBAC chart match the expected resources.
+func ValidateGardenletRBACChartResources(ctx context.Context, c client.Client) {
```

Comment:
See above, I really don't understand why we would need such validation. If the resources are missing then it will fail anyways.

Reply:
>  If the resources are missing then it will fail anyways.

Sorry, I do not understand that argument. This test is to make sure that the required RBAC rules are actually created and that the RBAC rules have not been modified in the helm chart. That is not different from any other unit test?

Reply:
Yeah, I guess it boils down to what we already discuss in another comment, i.e., that the RBAC roles don't be long into the gardenlet component. The test itself makes sense if we introduce the Gardener control plane component, so you can keep/store the code somewhere so that we can later re-use it.